### PR TITLE
Added custom carousel slide example

### PR DESCRIPTION
**Pull request by ThusharaJ07**
JIRA: CIBDESIGNSYSTEMJIRA-468

### CODE REVIEW

File: site/docs/components/carousel/examples.mdx [hunk 1]
```diff
@@ -82,3 +82,9 @@ The Embla API enables programmatic control, such as the visible slide.
 </Callout>
 
 <LivePreview componentName="carousel" exampleName="ControlledCarousel" />
+
+## Custom slide
+
+You can render fully custom slide content, such as `Avatar` components, by composing your own slide elements inside `CarouselSlides`.
```
Comment (mark-tate) [inline site/docs/components/carousel/examples.mdx]: Fully custom slides ?  any concerns about convergence @ivan-calderon ?

I'm not sure why we would have a carousel of Avatars, feels like an edge case rather than something that adds value to the doc. Typically examples are real-world use-cases that you copy and paste into code, anything less than that, should stay in Storybook as a test rig or edge-case example.

I think maybe we go with just an example in Storybook and prove it's possible without UX issues (see other comment) then update our internal skill to handle migration of avatar edge cases, based on the proven storybook code.

If this stays...
At least we would say, what the use-case is - is it for bios or something ?

**Comment by mark-tate** [comment]
This code has a UX issue, click on any avatar and it scroll to that avatar before it allows you to drag it.

File: site/docs/components/carousel/examples.mdx (hunk 1 above)
Comment (ThusharaJ07) [reply site/docs/components/carousel/examples.mdx]: I can move it storybook to test it out.

Comment (wemmyo) [reply site/docs/components/carousel/examples.mdx]: @ivan-calderon Also this looks like some variant of Avatar group which we decided not to support unless the avatars overlap with each other within group. So might be worth aligning on the design side

**Comment by ThusharaJ07** [comment]
Fixed another story based issue:
Card-based Carousel stories were hiding Link actions because the CSS rule relied on a class (.carouselSlideIsSnapped) that was never added to the DOM. The [embla-carousel-class-names] plugin — which produces that class — wasn't registered on any story, even though it is used correctly in the site docs example.

File: packages/embla-carousel/stories/carousel.stories.tsx [hunk 2]
```diff
@@ -37,6 +40,7 @@ const CarouselCardExample: StoryFn<CarouselProps & { ariaVariant: string }> = ({
     <Carousel
       aria-labelledby={`${carouselId}-title`}
       className="carousel"
+      emblaPlugins={[ClassNames({ snapped: "carouselSlideIsSnapped" })]}
```
Comment (joshwooding) [inline packages/embla-carousel/stories/carousel.stories.tsx:43]: Is this actually used?

**Review by joshwooding — DISMISSED** [review]

**Comment by joshwooding** [comment]
Waiting for @jake-costa's review

**Review by joshwooding — DISMISSED** [review]

**Review by joshwooding — APPROVED** [review]

**Review by wemmyo — APPROVED** [review]
